SonarQube.Client/Models/ServerSentEvents/SSEStreamReader.cs

```diff
@@ -0,0 +1,113 @@
+﻿/*
+ * SonarQube Client
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Threading;
+using System;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using SonarQube.Client.Models.ServerSentEvents.ClientContract;
+using SonarQube.Client.Models.ServerSentEvents.ServerContract;
+using System.Collections.Generic;
+using System.Threading.Channels;
+using SonarQube.Client.Logging;
+
+namespace SonarQube.Client.Models.ServerSentEvents
+{
+    public interface ISSEStreamReader
+    {
+        /// <summary>
+        /// Will block the calling thread until an event exists or the connection is closed.
+        /// Can throw an exception if the event is not a valid <see cref="IServerEvent"/>
+        /// </summary>
+        Task<IServerEvent> GetNextEventOrNullAsync();
```

Comment:
Under what circumstances will the method return `null`? When the channel is closed and there are no more events?

Reply:
Alternative name - `ReadAsync`?